app/jobs/notify_compatibles_job.rb

```diff
@@ -0,0 +1,9 @@
+class NotifyCompatiblesJob < ApplicationJob
+  queue_as :default
+ 
+  def perform(target)
+    list_id_onesignal = User.find(target.compatible_targets.pluck(:user_id)).pluck(:id_onesignal)
```

Comment:
Can't you do something like:
`list_id_onesignal = target.compatible_targets.joins(:user).pluck('users.id_onesignal')`